templates/cli/lib/questions.js.twig

```diff
@@ -768,7 +772,66 @@ const questionsMfaChallenge = [
     }
 ];
 
+const questionsRunFunctions = [
+    {
+        type: "list",
+        name: "function",
+        message: "Which function would you like to develop locally?",
+        validate: (value) => validateRequired('function', value),
+        choices: () => {
+            let functions = localConfig.getFunctions();
+            if (functions.length === 0) {
+                throw new Error("No functions found in the current directory.");
+            }
+            let choices = functions.map((func, idx) => {
+                return {
+                    name: `${func.name} (${func.$id})`,
+                    value: func.$id
+                }
+            })
+            return choices;
+        }
+    },
+    {
+        type: "input",
+        name: "port",
+        message: 'Which port would you like function to listen on?',
+        default: async () => {
+            let port = 3000;
```

Comment:
we should only show this prompt when the port 3000 is not available right? 
you can use the `when` condition of inquirer to check this

Reply:
Well we could automatically also pick 3001 because its available. Should we not ask this question at all, and just tell them to use `--port` if they want to change it?